Домашнее задание JS

AlexeyMoseev opened the pull request:
1. Реализовал получение данных на странице списка
2. Реализовал получение данных на странице карточки
3. Логику по получению данных вынес в отдельные файлы, назвал cardApi - для карточки товара, listApi - для списка товаров
4. Добавил интерактивность счетчику в карточке товара

AlexeyMoseev committed:
1. Реализовал получение данных на странице списка
2. Реализовал получение данных на странице карточки
3. Логику по получению данных вынес в отдельные файлы, назвал cardApi - для карточки товара, listApi - для списка товаров
4. Добавил интерактивность счетчику в карточке товара

AlexeyMoseev committed:
Реализовал поиск по свойству "name" на странице списка товаров

Roman-Maksimov commented on index.html:
```diff
@@ -30,7 +32,7 @@
 					</svg>
 				</button>
-				<input class="search-input" placeholder="Search products" />
+				<input id="1" class="search-input" placeholder="Search products" />
```
Про id я уже писал, такое имя формально хоть и считается валидным, стоит придерживаться соглашений XHTML. Это поможет избежать множества неприятных частых случаев в будущем. Так же, любое название переменной, id, класса должно нести смысловую нагрузку, чтобы было понятно с первого взгляда, о чем речь и что хранится в этой переменной.

Roman-Maksimov commented on js/cardApi.js:
```diff
@@ -0,0 +1,61 @@
+			if (this.status == 200) callback(JSON.parse(this.responseText))
+		}
+	}
+	xhr.send(null)
```
Зачем `null`?

Roman-Maksimov commented on js/cardApi.js:
```diff
@@ -0,0 +1,61 @@
+	xhr.send(null)
+}
+
+ajax('GET', 'http://localhost:3006/item/' + id, function (data) {
```
в ES6+ рекомендуется использовать шаблонный литерал
```javascript
`http://localhost:3006/item/${id}`
```

Roman-Maksimov commented on js/cardApi.js:
```diff
@@ -0,0 +1,61 @@
+			)
+		}
+	} catch (e) {
+		console.log(e)
```
Раз уж это ошибка, то `console.error(e)`

Roman-Maksimov commented on js/cardApi.js:
```diff
@@ -0,0 +1,61 @@
+
+		//Добавление фото
+		const photo = document.getElementsByClassName('detailed-image')[0]
+		photo.src = 'http://localhost:3006/' + data.content.picture.path
```
Шаблонный литерал

Roman-Maksimov commented on js/counter.js:
```diff
@@ -0,0 +1,12 @@
+let countInput = document.getElementsByClassName('count')[0];
+let remove = document.getElementsByClassName('remove')[0];
+let add = document.getElementsByClassName('add')[0];
+let units = countInput.value.replace(/\d/g, '');
```
Я не вижу, где вы переопределяете все эти четыре переменные. `let` предполагает, что вы будете где-то дальше их переназначать, в противном случае надо использовать `const`.

Roman-Maksimov commented on js/listApi.js:
```diff
@@ -0,0 +1,54 @@
+			photo.alt = data.content[i].picture.alt
+		}
+	} catch (e) {
+		console.log(e)
```
console.error

Roman-Maksimov commented on js/listApi.js:
```diff
@@ -0,0 +1,54 @@
+			value.innerHTML = data.content[i].price.value
+
+			const photo = document.getElementsByClassName('image')[i]
+			photo.src = 'http://localhost:3006/' + data.content[i].picture.path
```
шаблонный литерал

Roman-Maksimov reviewed (changes requested):
- Исправьте везде не тождественные сравнения
- Нет ни одного конца строки (символ `;`). Он хоть и не обязателен, но крайне рекомендуется к использованию.

AlexeyMoseev committed:
1. Исправил все не тождественные сравнения
2. Везде добавил символы конца строки (символ ;)
3. Убрал id, который не нес никакой смысловой и информационной нагрузки
4. Убрал null в методе xhr.send()
5. Сделал шаблонные литералы, в местах где идет обращение к серверу
6. В местах где идет сообщение об ошибке написал console.error(e) вместо console.log(e)
7. Исправил let на const там где это было необходимо
